src/urllib3/connection.py

```diff
@@ -709,7 +717,7 @@ def connect(self) -> None:
 
                 self._tunnel()
                 # Override the host with the one we're requesting data from.
-                server_hostname = self._tunnel_host
+                server_hostname = self._tunnel_host  # type: ignore[assignment]
```

Comment:
What do you think about casting `self._tunnel_host` to `str` instead of ignoring the type?

```suggestion
                server_hostname = typing.cast(str, self._tunnel_host)
```